assets/js/admin/course-edit.js

```diff
@@ -1,36 +1,59 @@
-/**
- * It enables or disables the legacy meta boxes.
- *
- * @param {boolean} enable Whether enable or disable.
- */
-window.sensei_toggleLegacyMetaboxes = ( enable ) => {
-	// Side-effect - Prevent submit modules.
-	document
-		.querySelectorAll( '#module_course_mb input' )
-		.forEach( ( input ) => {
-			input.disabled = ! enable;
-		} );
-
-	const editPostSelector = wp.data.select( 'core/edit-post' );
-	const editPostDispatcher = wp.data.dispatch( 'core/edit-post' );
+import { select, dispatch } from '@wordpress/data';
 
-	const legacyMetaboxes = [
-		'meta-box-course-lessons',
-		'meta-box-module_course_mb',
-		'meta-box-course-video',
+( () => {
+	const COURSE_OUTLINE_NAME = 'sensei-lms/course-outline';
+	const OTHER_COURSE_BLOCKS = [
+		'sensei-lms/button-take-course',
+		'sensei-lms/button-contact-teacher',
+		'sensei-lms/course-progress',
 	];
 
-	legacyMetaboxes.forEach( ( legacyMetabox ) => {
-		if (
-			enable !== editPostSelector.isEditorPanelEnabled( legacyMetabox )
-		) {
-			editPostDispatcher.toggleEditorPanelEnabled( legacyMetabox );
-		}
-	} );
-};
+	const blockEditorSelector = select( 'core/block-editor' );
+	const editPostSelector = select( 'core/edit-post' );
+	const editPostDispatcher = dispatch( 'core/edit-post' );
+
+	/**
+	 * Toggle meta boxes depending on the blocks.
+	 */
+	window.sensei_toggleLegacyMetaboxes = () => {
```

Comment:
I am sure that there is a reason behind it but why is this half snake case?

Reply:
Honestly, I don't remember exactly why I named like this, but probably I created the function, and after I just added the `sensei_` prefix because it's global forgetting to change the rest. Checking the other global variables we have, they are all snake case, so I changed for that too. Good catch! https://github.com/Automattic/sensei/commit/29d006e161ae520e10c12c86ab94eca4f6571c39